ui/packages/catalog-ui-search/src/main/webapp/react-component/container/metacard-interactions/index.tsx

```diff
@@ -298,114 +225,194 @@ const createResultActionsExportRegion = (model: Model) =>
+              click the Settings icon, select Hidden, then choose to unhide the record."
+        icon="fa fa-eye-slash"
+        onClick={() => handleHide(props)}
+      />
+    )
+  }
+}
+
+const ExpandMetacard = (props: any) => {
+  const isRouted = router && router.toJSON().name === 'openMetacard'
+
+  if (isRouted || props.model.length > 1) {
+    return null
+  }
+
+  return (
+    <MetacardInteraction
+      text="Expand Metacard View"
+      help="Takes you to a
+            view that only focuses on this particular result.  Bookmarking it will allow
+            you to come back to this result directly."
+      icon="fa fa-expand"
+      onClick={() => handleExpand(props)}
+    />
+  )
+}
+
+type State = {
+  blacklisted: Boolean
+  model: any
+}
+
+const interactions = plugin([
+  AddToList,
+  BlacklistToggle,
+  ExpandMetacard,
+  Divider,
+  DownloadProduct,
+  CreateLocationSearch,
+  ExportActions,
+])
+
+class MetacardInteractions extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props)
+    this.state = {
+      model: null,
+      blacklisted: false,
+    }
+  }
+  componentDidMount = () => {
+    const setState = (model: Model) => this.setState({ model: model })
 
     this.props.listenTo(
       this.props.model,
       'change:metacard>properties',
       setState
     )
+
     this.props.listenTo(
       user
         .get('user')
         .get('preferences')
         .get('resultBlacklist'),
       'add remove update reset',
-      toggleIsBlacklisted
+      () => this.setState({ blacklisted: isBlacklisted(this.props.model) })
     )
   }
 
   render = () => (
     <>
-      <MarionetteRegionContainer
-        data-help="Add the result to a list."
-        className="metacard-interaction interaction-add"
-        view={createAddRemoveRegion(this.props.model)}
-        viewOptions={{ model: this.props.model }}
-      />
-      <View
-        handleCreateSearch={() =>
-          withCloseDropdown(this.props, handleCreateSearch)
-        }
-        handleDownload={() => withCloseDropdown(this.props, handleDownload)}
-        isRemoteResourceCached={isRemoteResourceCached(this.props.model)}
-        withCloseDropdown={handler => withCloseDropdown(this.props, handler)}
-        viewModel={viewModelFromProps(this.props)}
-      />
-      <MarionetteRegionContainer
-        data-help="Opens the available actions for the item."
-        className="metacard-interaction interaction-actions-export composed-menu"
-        view={createResultActionsExportRegion(this.props.model)}
-        viewOptions={{ model: this.props.model }}
-      />
-      {this.props.extensions && (
-        <MarionetteRegionContainer
-          className="composed-menu interaction-extensions"
-          view={this.props.extensions}
-          viewOptions={{ mode: this.props.model }}
-        />
-      )}
+      {interactions.map((Component: any, i: number) => {
+        return (
+          <Component
+            key={i}
```

Comment:
This isn't a safe way to set keys.  https://reactjs.org/docs/lists-and-keys.html

Reply:
Discussed, this list won't be changing.